src/openrct2-ui/windows/Ride.cpp

```diff
@@ -5031,23 +5017,19 @@ static void window_ride_music_resize(rct_window* w)
     window_set_resize(w, 316, 81, 316, 81);
 }
 
-static std::optional<size_t> GetMusicStyleOrder(ObjectEntryIndex musicObjectIndex)
+static std::string GetMusicString(ObjectEntryIndex musicObjectIndex)
 {
     auto& objManager = GetContext()->GetObjectManager();
     auto musicObj = static_cast<MusicObject*>(objManager.GetLoadedObject(ObjectType::Music, musicObjectIndex));
 
-    // Get the index in the order list
-    auto originalStyleId = musicObj->GetOriginalStyleId();
-    if (originalStyleId.has_value())
+    auto name = musicObj->NameStringId;
+    if (name != '\0')
     {
-        auto it = std::find(std::begin(MusicStyleOrder), std::end(MusicStyleOrder), originalStyleId.value());
-        if (it != std::end(MusicStyleOrder))
-        {
-            return std::distance(std::begin(MusicStyleOrder), it);
-        }
+        static char buffer[256];
+        format_string(buffer, 256, name, static_cast<void*>(&name));
```

Comment:
Just looking at the overloads and format_string has an overload that returns a std::string. Just use that and avoid all the buffer stuff.